api/core/v1alpha1/digest.go

```diff
@@ -22,7 +21,7 @@ func init() {
 }
 
 func (d *Digest) ToString() string {
```

Comment:
maybe we rename `ToString -> Encode`, and `FromString -> Decode` which will construct the digest into `<digest-type>:<digest-value>` string and read from it

Reply:
completely agree!